[12.0][MIG] website_event_filter_selector: Migration to v12

cristinamartinrod opened the pull request:
cc @Tecnativa

yajo committed:
[8.0][website_event_filter_selector] Filter events with selectors.

You can use selectors above the event list in the website, instead of the left column that there is right now.

Store user customizations in separate template.

Acknowledge that you cannot hide only the date filter.

Remove search button; search on selection change.

Fix bug where date did not match what was being filtered.

[FIX][8.0][website_event_filter_selector] Do not die without types or countries

This basically fixes #44 and a similar bug detected that would happen when no countries were found.

It just adds a `t-if` to avoid possible failures, [borrowing the idea from core `website_event` module](https://github.com/odoo/odoo/blob/8.0/addons/website_event/views/website_event.xml#L167).

[8.0][IMP][website_event_filter_selector] Add city filter for events.

Since this includes new logic, it adds also a template for left-column filters.

With this patch, you can now filter events depending on their cities, not only on their countries.

This also includes some modifications that will make filters occupy always a full bootstrap row, no matter if there are 1, 2, 3 or 4 filters, to avoid ugly layouts.

pedrobaeza committed:
[IMP] website_event_filter_selector

[IMP] website_event_filter_selector: Spanish translation

[IMP] website_event_filter_selector: German translation

[FIX] website_event_filter_selector: Fix date selection

[FIX] website_event_filter_selector: Fix city count

[FIX] website_event_filter_selector: Don't lose context with new obj

yajo committed:
[9.0][website_event_filter_selector] Migrate.

Relicensed to LGPL, new JS API.

Add test. Fix errors found with it.

Go 100% coverage. Support online events.

[FIX][website_event_filter_selector] Fix Travis build

Builds failed on 2017-03-31 because the tour was expecting to see USA event starting in this month, but next day was another month.

Well, in short, this fix makes the tour work on any day.

yajo committed:
[10.0][MIG][website_event_filter_selector]

Includes update to use new tours api.

Patch event counts when a city is searched.

tarteo committed:
[MIG] website_event_filter_selector: Migration to 11.0

[IMP] Use new README approach

[IMP] Removed the use of @class

[FIX] Column classes

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

Commit:
[12.0][MIG] website_event_filter_selector

pedrobaeza commented:
Tests are not correct.

yajo commented on website_event_filter_selector/tests/test_ui.py:
```diff
@@ -0,0 +1,49 @@
+            },
+        }
+        with self.cursor() as cr:
+            env = self.env(cr)
```
This hack is no longer needed in v12. Remove the `with` block and use `self.env` instead of `env`.

yajo commented on website_event_filter_selector/__manifest__.py:
```diff
@@ -0,0 +1,30 @@
+    ],
+    # 'images': [
+    #     'static/description/website_event_filter_selector.jpg',
+    # ],
```
:fire: remove this.

yajo reviewed (changes requested):
There's one change to do. I'm not sure if it will fix travis...

rafaelbn commented:
@cristinamartinrod travis is :red_circle: because:

```
2019-03-04 12:15:46,089 6475 ERROR openerp_test odoo.addons.website_event_filter_selector.tests.test_ui: FAIL
2019-03-04 12:15:46,089 6475 INFO openerp_test odoo.addons.website_event_filter_selector.tests.test_ui.UICase: Closing chrome headless with pid 6492
2019-03-04 12:15:46,090 6475 INFO openerp_test odoo.addons.website_event_filter_selector.tests.test_ui.UICase: Terminating chrome headless with pid 6492
2019-03-04 12:15:46,101 6475 INFO openerp_test odoo.addons.website_event_filter_selector.tests.test_ui.UICase: Removing chrome user profile "/tmp/tmph2r37vb4_chrome_odoo"
2019-03-04 12:15:46,102 6475 INFO openerp_test odoo.addons.website_event_filter_selector.tests.test_ui: ======================================================================
2019-03-04 12:15:46,102 6475 ERROR openerp_test odoo.addons.website_event_filter_selector.tests.test_ui: FAIL: test_ui_website (odoo.addons.website_event_filter_selector.tests.test_ui.UICase)
2019-03-04 12:15:46,102 6475 ERROR openerp_test odoo.addons.website_event_filter_selector.tests.test_ui: ` Test frontend tour.
2019-03-04 12:15:46,103 6475 ERROR openerp_test odoo.addons.website_event_filter_selector.tests.test_ui: Traceback (most recent call last):
2019-03-04 12:15:46,103 6475 ERROR openerp_test odoo.addons.website_event_filter_selector.tests.test_ui: `   File "/home/travis/build/OCA/event/website_event_filter_selector/tests/test_ui.py", line 48, in test_ui_website
2019-03-04 12:15:46,103 6475 ERROR openerp_test odoo.addons.website_event_filter_selector.tests.test_ui: `     ready="odoo.__DEBUG__.services['web_tour.tour']"
2019-03-04 12:15:46,103 6475 ERROR openerp_test odoo.addons.website_event_filter_selector.tests.test_ui: `   File "/home/travis/odoo-12.0/odoo/tests/common.py", line 923, in browser_js
2019-03-04 12:15:46,103 6475 ERROR openerp_test odoo.addons.website_event_filter_selector.tests.test_ui: `     self.assertTrue(self.browser._wait_ready(ready), 'The ready "%s" code was always falsy' % ready)
2019-03-04 12:15:46,104 6475 ERROR openerp_test odoo.addons.website_event_filter_selector.tests.test_ui: ` AssertionError: False is not true : The ready "odoo.__DEBUG__.services['web_tour.tour'].tours.website_event_filter_selector.ready" code was always falsy
```

yajo commented:
Right now tests are failing. See: https://travis-ci.org/OCA/event/jobs/503523853#L1537
Possibly you need to modify the tour.

yajo committed:
[FIX] website_event_filter_selector: Update tour

- Use own demo data instead of upstream.
- Fix some JS code that wasn't working as expected.
- Remove dependency on left column; just test own filters widget.

yajo reviewed (approved):
Tests fixed.
